test/extended/util/cluster/cluster.go

```diff
@@ -38,6 +38,10 @@ type ClusterConfiguration struct {
 	// Disconnected is set for test jobs without external internet connectivity
 	Disconnected bool
 
+	// SingleReplicaTopology is set for disabling disruptive tests or tests
+	// that require high availability
+	SingleReplicaTopology bool
```

Comment:
Maybe reuse the MultiMaster field in this struct, but change it so it gets set as `*state.ControlPlaneTopology == configv1.SingleReplicaTopologyMode,` rather than `len(state.Masters.Items) > 1,`

Reply:
there is one usage for the multimaster field, and I do not intend to break it:
https://github.com/openshift/origin/blob/6adc3079a51f5ca677cdf445879a9feb407a22ac/vendor/k8s.io/kubernetes/test/e2e/framework/nodes_util.go#L119

Reply:
I didn't say remove the Multimaster field, I said reuse the MultiMaster field

Reply:
You can assert that indeed `!(len(state.Masters.Items) > 1)` whenever  `*state.ControlPlaneTopology == configv1.SingleReplicaTopologyMode` to verify the definitions have no diverged in any way

Reply:
also it's not conforming to the field name
you were right the first time, I need to use the topology not the master-counter field :)

Reply:
I didn't say you shouldn't use the topology, I said we don't need a new field for it, i.e.:

```go
	config := &ClusterConfiguration{
		MultiMaster:           *state.ControlPlaneTopology != configv1.SingleReplicaTopologyMode,
		MultiZone:             zones.Len() > 1,
		Zones:                 zones.List(),
	}
```

Reply:
it's not the same thing
it seems like they use multimaster to distinguish between multi-zone and multi-regional cases on cloud, and it's not the same as cluster topology 
I prefer creating a new field with the right meaning to its value, and see afterwards if it makes sense for the old field to be merged to the new one, because right now I doubt it

Reply:
Okay, not insisting, just making sure my comments are understood